trio/_abc.py

```diff
@@ -252,10 +254,10 @@ async def aclose(self):
 
         """
 
-    async def __aenter__(self):
+    async def __aenter__(self) -> AsyncResource:
         return self
 
-    async def __aexit__(self, *args):
+    async def __aexit__(self, *args: object) -> None:
```

Comment:
I think fixing this signature would be nice!

Reply:
I'll do a big search&replace on all `__exit__` and `__aexit__` in all files, so I won't have to juggle the trio of types and names more. I'll also standardize them to be the same.

Reply:
> This signature is fine, given users shouldn’t be calling directly. It accepts the parameters, and means we don’t need to import `types`.

I'm hiding the `types` import behind `if TYPE_CHECKING`, so if anything it saves importing `typing` - which pretty much all of the files will be importing anyway.